src/plugins/data_source/common/data_sources/error.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ResponseError } from '@opensearch-project/opensearch/lib/errors';
+import { OsdError } from '../../../opensearch_dashboards_utils/common';
+
+export class DataSourceError extends OsdError {
+  // must have statusCode to avoid route handler in search.ts to return 500
+  statusCode: number;
+  body: any;
```

Comment:
nit: use `let` for consistent code

Reply:
Hi, this chunk of code only remove, not make any update

Reply:
Yes, I understand that you removed this code from [src/plugins/data_source/server/lib/error.ts](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6903/files#diff-ef69cccd90d655f9875d637d8dba88db1a21197b585ec2b0ee66d44ba9a67a6b) and moved to [src/plugins/data_source/common/data_sources/error.ts](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6903/files#diff-4fbcbc9f0b4b7cfd2f8f730e176a9cc369b04897ab6a999034f7e826784eb9dc). But to be consistent I am suggesting to uset `let` to declare variable here

Reply:
this is a class, we cannot use the let here, we better to use private/public/static etc accessors for the fields within the class